### PR TITLE
Adapt RHTS to work with Python 3

### PR DESCRIPTION
**Pull request by renanrodrigo**
- Add python-six for cross-version compatibility
- Change preamble of executable files to detect the Python version
correctly
- Change the code base to have compatible statements and calls
- Standardize tabs and spaces in files

Bug: 1750646

Change-Id: Ifd24e194776404a749a72f44fa5daef5d239eac4

### CODE REVIEW

**Comment by StykMartin** [comment]
@mdujava please review

**Review by StykMartin — CHANGES_REQUESTED** [review]
Let's finalize this in this PR. We need one more commit here. To update the spec file.

**Comment by StykMartin** [comment]
Tests are failing

File: bin/rhts-abort [hunk 1]
```diff
@@ -41,7 +53,7 @@ def abortJob(server, jobid):
             resp = server.watchdog.abortJob(jobid)
             break
         except:
-            print "Unable to connect to server, sleeping 5 seconds..."
+            print ("Unable to connect to server, sleeping 5 seconds...")
```
Comment (MartinBasti) [inline bin/rhts-abort]: :eyes:

Comment (renanrodrigo) [reply bin/rhts-abort]: I feel bad about the eyes emoji, my mom will receive a report about this.
She loves me.

File: bin/rhts-abort [hunk 2]
```diff
@@ -76,15 +88,15 @@ def main():
     recipeid = None
     recipetestid = None
 
-    if ('RESULT_SERVER' in os.environ.keys()):
+    if ('RESULT_SERVER' in list(os.environ.keys())):
```
Comment (MartinBasti) [inline bin/rhts-abort]: no need for `list` in this case

File: bin/rhts-db-submit-result [hunk 3]
```diff
@@ -260,7 +278,7 @@ class AuResult(object):
         self._success = success
         self._avc = avc
 
-    def __nonzero__(self):
+    def __bool__(self):
```
Comment (MartinBasti) [inline bin/rhts-db-submit-result:281]: for py2/3 compat:
```python
if six.PY2:
    __nonzero__ = __bool__
```

File: bin/rhts-recipe-sync-block [hunk 4]
```diff
@@ -96,9 +109,9 @@ def usage():
 def main():
     global result_server, hostname, recipesetid
 
-    if ('RESULT_SERVER' in os.environ.keys()):
+    if ('RESULT_SERVER' in list(os.environ.keys())):
```
Comment (MartinBasti) [inline bin/rhts-recipe-sync-block]: :eyes:

File: bin/rhts-recipe-sync-set [hunk 5]
```diff
@@ -60,24 +72,24 @@ def sync_set(state):
...
 def usage():
-    print USAGE_TEXT
+    print (USAGE_TEXT)
     sys.exit(-1)
 
 def main():
     global result_server, hostname, recipesetid
 
-    if ('RESULT_SERVER' in os.environ.keys()):
+    if ('RESULT_SERVER' in list(os.environ.keys())):
```
Comment (MartinBasti) [inline bin/rhts-recipe-sync-set]: :eyes:

File: bin/rhts-submit-log [hunk 6]
```diff
@@ -47,9 +59,9 @@ def main():
     result_server = ''
     recipetestid = ''
 
-    if ('RESULT_SERVER' in os.environ.keys()):
+    if ('RESULT_SERVER' in list(os.environ.keys())):
```
Comment (MartinBasti) [inline bin/rhts-submit-log]: :eyes:

File: bin/rhts-sync-block [hunk 7]
```diff
@@ -105,11 +118,11 @@ def usage():
 def main():
     global result_server, hostname, recipesetid, testorder
 
-    if ('RESULT_SERVER' in os.environ.keys()):
+    if ('RESULT_SERVER' in list(os.environ.keys())):
```
Comment (MartinBasti) [inline bin/rhts-sync-block]: :eyes:

File: python-modules/rhts/rhts_workflow.py [hunk 8]
```diff
@@ -69,14 +69,14 @@ def __deepcopy__(self, memo):
 		return self
 
 	def getTest(self,testname):
-		if testname not in self.testcache.keys():
+		if testname not in list(self.testcache.keys()):
```
Comment (MartinBasti) [inline python-modules/rhts/rhts_workflow.py]: :eyes:

File: python-modules/rhts/rhts_workflow.py [hunk 9]
```diff
@@ -272,46 +271,46 @@ def driverdisk(self, dd):
 
 	def toxml(self):
 		doc = xml.dom.minidom.Document()
```
Comment (MartinBasti) [inline python-modules/rhts/rhts_workflow.py]: :eyes:  tabs?

File: python-modules/rhts/rhts_workflow.py [hunk 10]
```diff
@@ -272,46 +271,46 @@ def driverdisk(self, dd):
...
 			try:
-				urllib2.urlopen("%s/repodata/repomd.xml" % repo)
-			except urllib2.URLError,e:
-				print "Warning: %s - double-check the Yum repo." % e
+				urllib.request.urlopen("%s/repodata/repomd.xml" % repo)
+			except urllib.error.URLError as e:
+				print ("Warning: %s - double-check the Yum repo." % e)
 			recipe.appendChild(node("addrepo",repo))
-                for key in self.distro_properties.keys():
+		for key in list(self.distro_properties.keys()):
```
Comment (MartinBasti) [inline python-modules/rhts/rhts_workflow.py]: :eyes:

File: bin/rhts-db-submit-result (hunk 3 above)
Comment (MartinBasti) [reply bin/rhts-db-submit-result:281]: still valid

**Comment by StykMartin** [comment]
I would rate this 10/10 force push fiesta

**Review by MartinBasti — APPROVED** [review]

**Review by StykMartin — APPROVED** [review]

File: bin/rhts-recipe-sync-set [hunk 11]
```diff
@@ -60,20 +72,20 @@ def sync_set(state):
...
 
 def usage():
-    print USAGE_TEXT
+    print(USAGE_TEXT)
     sys.exit(-1)
 
 def main():
     global result_server, hostname, recipesetid
 
-    if ('RESULT_SERVER' in os.environ.keys()):
```
Comment (cbouchar) [inline bin/rhts-recipe-sync-set]: Are you sure you intended to remove 'in'?  I'm thinking you wanted to remove 'list' from previous changes and took in with you.  You didn't do it for RECIPESETID and HOSTNAME?????

File: bin/rhts-sync-block [hunk 12]
```diff
@@ -135,19 +148,19 @@ def main():
             timeout = datetime.datetime.now() + datetime.timedelta(seconds=int(val))
 
     machines = args[0:]
-    machines = filter(isset,machines)
+    machines = list(filter(isset,machines))
```
Comment (cbouchar) [inline bin/rhts-sync-block:151]: You sure this list should stay since you've removed it elsewhere.

**Review by cbouchar — CHANGES_REQUESTED** [review]
These files in gerrit have changes but none here in github.  Some of gerrit changes look good.  
rhts_options.py
rhts_workflow.py
testinfo.py

File: bin/rhts-sync-block (hunk 12 above)
Comment (MartinBasti) [reply bin/rhts-sync-block:151]: This one is correct, in py3 `filter` works as generator so we have to explicitly create list from it

File: python-modules/rhts/__init__.py [hunk 13]
```diff
@@ -170,14 +173,14 @@ def uploadWrapper(session, localfile, recipetestid, name=None, callback=None, bl
...
-                    print "uploadFile(%r,%r,%r,%r,%r,...)" %(recipetestid,name,sz,digest,offset)
+                    print("uploadFile(%r,%r,%r,%r,%r,...)" %(recipetestid,name,sz,digest,offset))
                 if callMethod(session, 'results.uploadFile', recipetestid, name, sz, digest, offset, data):
                     break
                 if tries <= retries:
                     tries += 1
                     continue
                 else:
-                    raise GenericError, "Error uploading file %s, offset %d" %(name, offset)
+                    raise GenericError ("Error uploading file %s, offset %d" %(name, offset))
```
Comment (MartinBasti) [inline python-modules/rhts/__init__.py]: Extra whitespace

File: python-modules/rhts/timeout_xmlrpclib.py [hunk 14]
```diff
@@ -36,7 +36,7 @@ class TimeoutTransport(Transport):
 
    def make_connection(self, host):
        if sys.version_info[:2] < (2, 7):
-           conn = TimeoutHTTP(host)
+           conn = TimeoutHTTPConnection(host)
```
Comment (MartinBasti) [inline python-modules/rhts/timeout_xmlrpclib.py]: Wrong indentation

File: python-modules/rhts/__init__.py [hunk 15]
```diff
@@ -170,14 +173,14 @@ def uploadWrapper(session, localfile, recipetestid, name=None, callback=None, bl
             tries = 0
```
Comment (MartinBasti) [inline python-modules/rhts/__init__.py:175]: this function (line 149) contains `file` keyword, this is not gonna work in py3

**Review by MartinBasti — CHANGES_REQUESTED** [review]
Sorry, I'll do better review next time

File: bin/rhts-recipe-sync-set (hunk 11 above)
Comment (renanrodrigo) [reply bin/rhts-recipe-sync-set]: Yes, thanks, it was a mistake...... Will fix it.....

**Comment by renanrodrigo** [comment]
@cbouchar about the file changes: I can see it in gerrit and in my local diff, don't know why github is not presenting the changes properly :owl:

**Review by cbouchar — APPROVED** [review]
I'll trust you on those files that show not updated when they actually are.  It's weird that it even lists
those files.  Confusing.....

**Comment by StykMartin** [comment]
> I'll trust you on those files that show not updated when they actually are. It's weird that it even lists
> those files. Confusing.....

In case you would like to review it, you can download branch to your copy and verify changes.

**Review by StykMartin — CHANGES_REQUESTED** [review]
\__future__ import(s) are not at the beginning of the file. This will cause SyntaxError.

**Comment by cbouchar** [comment]
looking now.

**Comment by cbouchar** [comment]
All looks wells.

**Comment by StykMartin** [comment]
Added patches for spec file. We still need to resolve issues I pointed above.

1. Update imports. To be sure that \__future__ is at the beginning.
2. bytes/str seems to be not resolved

**Review by StykMartin — APPROVED** [review]

**Comment by StykMartin** [comment]
@renanrodrigo congratulation for your first merged contribution to Beaker :tada:!

**Pull request merged**